Gadget.Server/Hubs/GadgetHub.cs

```diff
@@ -1,99 +1,120 @@
+using System;
 using System.Collections.Generic;
 using System.Linq;
 using System.Threading.Tasks;
-using Gadget.Messaging;
-using Microsoft.AspNetCore.SignalR;
-using Microsoft.Extensions.Logging;
 
 namespace Gadget.Server.Hubs
 {
     public class GadgetHub : Hub
     {
-        private readonly IDictionary<Guid, ICollection<Service>> _agents;
-        private readonly IDictionary<string, Guid> _connectedClients;
+        private readonly IList<Agent> _agents;
         private readonly ILogger<GadgetHub> _logger;
 
-        public GadgetHub(IDictionary<Guid, ICollection<Service>> agents, IDictionary<string, Guid> connectedClients,
-            ILogger<GadgetHub> logger)
+
+        public GadgetHub(List<Agent> agents, ILogger<GadgetHub> logger)
         {
             _agents = agents;
-            _connectedClients = connectedClients;
             _logger = logger;
         }
 
         public override Task OnDisconnectedAsync(Exception exception)
         {
             var cid = Context.ConnectionId;
-            _connectedClients.Remove(cid);
+            _agents.Remove(_agents.Single(x => x.ConnectionId == cid));
             _logger.LogInformation($"Client {cid} has disconnected");
             return Task.CompletedTask;
         }
 
         public Task RegisterMachineReport(RegisterMachineReport registerMachineReport)
         {
             _logger.LogInformation($"Received new machine report from agent {registerMachineReport.AgentId}");
-            _agents[registerMachineReport.AgentId] = registerMachineReport.Services.ToList();
+            var agent = _agents.FirstOrDefault(x => x.MachineId == registerMachineReport.AgentId);
+            if (agent == null)
+            {
+                _agents.Add(new Agent(registerMachineReport.AgentId, Context.ConnectionId, registerMachineReport.Services));
+            }
+
             foreach (var service in registerMachineReport.Services)
+            {
                 _logger.LogInformation($"Service name : {service.Name} status : {service.Status}");
+            }
+
+            return Task.CompletedTask;
+        }
+
+        public Task MachineHealthCheck(MachineHealthDataModel model)
+        {
+            var agent = _agents.FirstOrDefault(x => x.MachineId == model.MachineId);
+            if (agent != null) agent.MachineHealthData = model;
 
             return Task.CompletedTask;
         }
 
         //TODO Remove from group on disconnect
         public Task RegisterDashboard(RegisterNewDashboard registerNewDashboard)
         {
-            _logger.LogCritical("HEj hej hej roman");
+            _logger.LogCritical($"HEj hej hej roman");
             var cid = Context.ConnectionId;
             Groups.AddToGroupAsync(cid, "dashboard");
             return Task.CompletedTask;
         }
 
-        public Task Register(RegisterNewAgent registerNewAgent)
+        public Task Register(RegisterNewAgent registerNewAgent) // dodać du klasę MachineHealthDataModel ?
```

Comment:
nad tym pomyslimy jeszcze, bo mozliwe ze lepiej bedzie aby to serwer healthcheckowal agentow kiedy jest mu to potrzebne